examples/notebooks/get-started/try-apache-beam-yaml.ipynb

```diff
@@ -0,0 +1,581 @@
+    "    input: FormatResults\n",
+    "    config:\n",
+    "      file_path_prefix: \"data/result-pipeline-01\"\n",
+    "      file_name_suffix: \".txt\"\n",
+    "'''\n",
+    "save_to_file(pipeline, 'pipeline-01.yaml')"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Each pipeline specification has to start with a `pipeline` key on the first line.\n",
+    "Following, there is a list of transforms, such as the first one:\n",
+    "```\n",
+    "  # Read input data. Each line from the csv file is a String.\n",
+    "  - type: ReadFromText\n",
+    "    name: InputText\n",
+    "    config:\n",
+    "      file_pattern: data/kinglear.txt\n",
+    "```\n",
+    "This one reads an input file for futher processing.\n",
+    "Remember that the indentation is important here - it specifies object hierarchy.\n",
+    "YAML also supports comments - everything after the `#` is always treated as a comment. Use them to improve readability.\n",
+    "\n",
+    "Each operation has to specify the `type` descriptor and other fields, such as `name` and other transform-specific parameters.\n",
+    "Take a look at the documentation for a list of available transforms and their parameters. # todo(yaml) add link\n",
+    "\n",
+    "To link two operations, use the `input` field, just like in the third operation:\n",
+    "```\n",
+    "  # Transforming each word to lower case and combining it with a '1'. Result of this step are pairs (word: 1).\n",
+    "  - type: PyMap\n",
+    "    name: PairWordsWith1\n",
+    "    input: FindWords\n",
+    "    config:\n",
+    "      fn: 'lambda word: (word, 1)'\n",
+    "```\n",
+    "The `input` refers to the name of another transform, the previous one in this case.\n",
+    "This particular operation takes `fn` (stands for function) as an argument. Currently only Python functions are supported.\n",
+    "For more complicated functions, you can take advantage of YAML's multiline feature, as you can see in the second operation:\n",
+    "```\n",
+    "  # Using a regex, we'll split the content of the message (one long string) into words (list of strings).\n",
+    "  - type: PyFlatMap\n",
+    "    name: FindWords\n",
+    "    input: InputText\n",
+    "    config:\n",
+    "      fn: |\n",
+    "        import re\n",
+    "        lambda line: re.findall(r\"[a-zA-Z]+\", line)\n",
+    "```\n",
+    "Here we had to import Python's regex package - `re`. To do that, we indicated a multiline string with `|` character and wrote the function in 2 lines.\n",
+    "\n",
+    "Let's run the pipeline executing the Python entry-point script (`apache_beam.yaml.main`) with our pipeline file as an argument:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "run('python -m apache_beam.yaml.main --pipeline_spec_file=pipeline-01.yaml')"
```

Comment:
I get an exception here:

ValueError: Invalid transform specification at "InputText" at line 4: PTransform.__init__() got an unexpected argument 'config'

(Previous steps seemed to work.)

Reply:
Hi Veronica! Sorry, I think you need to run `pip install -e .` in the sdks/python directory in the same python virtual environment as the notebook is running. 
This will not be necessary once the changes that this notebook relies on are released to the Python's central repository PyPi

Reply:
Should this be functional by now? :)
As a developer with minimal experience in pipelines i've also tried using the notebook (without changes whatsoever) and got an exception with the validator in yaml_transform.py: 
`jsonschema.exceptions.ValidationError: [.......] is not of type 'object' at line unknown`

Reply:
Hi @yardenbm, 
there were some changes in the yaml api, which make this notebook outdated. I'm updating it as we speak. It should be functional by the end of this week, and merged soon after. 
If you're looking for some examples on how to construct a pipeline, take a look at the Readme.md (https://github.com/apache/beam/tree/master/sdks/python/apache_beam/yaml#readme) and other .md files and unit tests in https://github.com/apache/beam/tree/master/sdks/python/apache_beam/yaml